xbmc/guilib/GUIControl.h

```diff
@@ -219,6 +219,7 @@ class CGUIControl
   virtual void SetInvalid() { m_bInvalidated = true; };
   virtual void SetPulseOnSelect(bool pulse) { m_pulseOnSelect = pulse; };
   virtual std::string GetDescription() const { return ""; };
+  virtual std::string GetDescriptionByIndex(int index) const { return ""; };
```

Comment:
xhaggi just because having the same method with different prototypes is ok for the compiler doesn't make it good practice. Naming a method like what it does just reads fine - and the byindex stuff is widly used in other apis. Not all of us devs use IDEs with intellisense or something like that which would make your approach easier to live with.
Also proposing to call a baseclass method without prefixing it with the baseclasses scope is bad practice - you read that code in a year and tell me that it doesn't look like recursion...